lib/options_validator.js

```diff
@@ -209,9 +209,20 @@ class OperationBuilder {
         validationOptions = { validationLevel: this.s.options.validationLevel };
       }
 
-      // validate options
-      const optionsIndex =
-        typeof args[args.length - 1] === 'function' ? args.length - 2 : args.length - 1;
+      const hasCallback = typeof args[args.length - 1] === 'function';
+      let optionsIndex = hasCallback ? args.length - 2 : args.length - 1;
+      if (optionsIndex < operationBuilder.requiredArity) {
```

Comment:
I'm not sure this is right. The algorithm we should be following for parsing parameters is basically:

```js
const callback = typeof args[args.length - 1] === 'function' ? args.pop() : whateverTheDefaultForCallbackIs;

if (args.length < arity || args.length > arity + 1) {
  // ERROR YOU HAVE THE WRONG # OF OPTIONS
}

const options = typeof args[arity] === object ? args.pop(): {};

args.push(options);
if (callback) {
  args.push(callback);
}
```

Basically, we would ensure the following:
+ We will assume a mistake in the following circumstances:
    + `args.length < arity`: Too few required args
    + `args.length > arity + 2`: Too many required args
    + `args.length === arity && typeof args[args.length - 1] === 'function'`: We assume the last arg is meant to be a callback, so there are too few required args
    + `args.length === arity + 2 && typeof args[args.length - 1] !== 'function'`: since last arg is not a callback, there are too many required args.
+ If someone passes `arity + 1` args, and the last arg is a callback, we will assume that they are omitting `options` and will provide a default
+ If someone passes `arity + 1` args, and the last arg is not a callback, we will assume that they are omitting `callback`, and that the last arg is `options`
+ If someone passes `arity + 2` args, and the last arg is a callback, we will assume that `args[arity]` is `options` and `args[arity + 1]` is `callback`